test/no-fn-reference-in-iterator.js

```diff
@@ -1,114 +1,212 @@
+					`foo.${methodName}((accumulator, element, index) => fn(accumulator, element, index))`,
+					`foo.${methodName}((accumulator, element, index, array) => fn(accumulator, element, index, array))`
+				]
+			})
+		),
+
+		// 2 arguments
+		...simpleMethods.map(
+			methodName => invalidTestCase({
+				code: `foo.${methodName}(fn, thisArgument)`,
+				methodName,
+				functionName: 'fn',
+				suggestions: [
+					`foo.${methodName}((element) => fn(element), thisArgument)`,
+					`foo.${methodName}((element, index) => fn(element, index), thisArgument)`,
+					`foo.${methodName}((element, index, array) => fn(element, index, array), thisArgument)`
+				]
+			})
+		),
+		...reduceLikeMethods.map(
+			methodName => invalidTestCase({
+				code: `foo.${methodName}(fn, initialValue)`,
+				methodName,
+				functionName: 'fn',
+				suggestions: [
+					`foo.${methodName}((accumulator, element) => fn(accumulator, element), initialValue)`,
+					`foo.${methodName}((accumulator, element, index) => fn(accumulator, element, index), initialValue)`,
+					`foo.${methodName}((accumulator, element, index, array) => fn(accumulator, element, index, array), initialValue)`
+				]
+			})
+		),
+
+		// `Boolean` is not ignored on `reduce` and `reduceRight`
+		...reduceLikeMethods.map(
+			methodName => invalidTestCase({
+				code: `foo.${methodName}(Boolean, initialValue)`,
+				methodName,
+				functionName: 'Boolean',
+				suggestions: [
+					`foo.${methodName}((accumulator, element) => Boolean(accumulator, element), initialValue)`,
+					`foo.${methodName}((accumulator, element, index) => Boolean(accumulator, element, index), initialValue)`,
+					`foo.${methodName}((accumulator, element, index, array) => Boolean(accumulator, element, index, array), initialValue)`
+				]
+			})
+		),
+
+		// Actual messages
 		{
 			code: 'foo.map(fn)',
-			errors,
-			output: 'foo.map(x => fn(x))'
-		},
-		{
-			code: 'foo.forEach(fn)',
-			errors,
-			output: 'foo.forEach(x => fn(x))'
+			errors: [
+				{
+					message: 'Do not pass function `fn` directly to `map()`.',
+					suggestions: [
+						{desc: 'Replace function `fn` with `(element) => fn(element)`.'},
+						{desc: 'Replace function `fn` with `(element, index) => fn(element, index)`.'},
+						{desc: 'Replace function `fn` with `(element, index, array) => fn(element, index, array)`.'}
+					]
+				}
+			]
 		},
 		{
-			code: 'foo.every(fn)',
-			errors,
-			output: 'foo.every(x => fn(x))'
+			code: 'foo.reduce(fn)',
+			errors: [
+				{
+					message: 'Do not pass function `fn` directly to `reduce()`.',
+					suggestions: [
+						{desc: 'Replace function `fn` with `(accumulator, element) => fn(accumulator, element)`.'},
+						{desc: 'Replace function `fn` with `(accumulator, element, index) => fn(accumulator, element, index)`.'},
+						{desc: 'Replace function `fn` with `(accumulator, element, index, array) => fn(accumulator, element, index, array)`.'}
```

Comment:
Is this message too much?

Reply:
Simplified a little bit ``Replace function `fn` with `… => fn(accumulator, element, index, array)`.``